lib/plugins/authldap/auth.php

```diff
@@ -463,9 +463,12 @@ protected function _constructPattern($filter) {
      * @return string
      */
     protected function _filterEscape($string) {
-        return preg_replace(
-            '/([\x00-\x1F\*\(\)\\\\])/e',
-            '"\\\\\".join("",unpack("H2","$1"))',
+        // see https://github.com/adldap/adLDAP/issues/22
+        return preg_replace_callback(
+            '/([\x00-\x1F\*\(\)\\\\])/',
+            function ($matches) {
+                return "\\".join("", unpack("H2", $matches[1]));
```

Comment:
Yes, I also wondered - this is quite confusing. The old code actually has 5 backslashs, which has to be an error, only 4 are meaningful: it's one backslash that is doubly escaped, once for the inner double quotes, once for the outer single quotes.

Removing the outer quotes from '"\\\\\"' yields "\\\", as can be verified through print('"\\\\\"').
"\\\" is equivalent to "\\", which cannot be easily verified, since print("\\\") or the like will give a parsing error.

I verified the equality of the old and the new string via evaluating the following expression, which gives TRUE:
preg_replace('/./e', '"\\\\\"', 'x') === preg_replace_callback('/./', function ($matches) { return "\\"; }, 'x');